### ISSUE TITLE
Failed to precompile devtools:devtools

### ISSUE THREAD

**Issue by Jomik** [flutter/devtools#2361]
Been trying a few things to get my devtools to run, but it "suddenly" wont work anymore. It has been a while since I used them, so not entirely sure when this happened..
I managed to get some output with the VSCode `Dart: Capture Logs` and also when trying `flutter pub cache repair`.

<details>
<summary>Cache repair output</summary>

```
Precompiling executables...
Failed to precompile devtools:devtools:
../../flutter/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart:276:33: Error: The method 'UnmodifiableListView' isn't defined for the class 'SpawnedProcess'.
 - 'SpawnedProcess' is from 'package:vm_service/src/dart_io_extensions.dart' ('../../flutter/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart').
Try correcting the name to the name of an existing method, or defining a method named 'UnmodifiableListView'.
  List<String> get arguments => UnmodifiableListView(_arguments);
                                ^^^^^^^^^^^^^^^^^^^^
../../flutter/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart:302:44: Error: The method 'UnmodifiableListView' isn't defined for the class 'SpawnedProcessList'.
 - 'SpawnedProcessList' is from 'package:vm_service/src/dart_io_extensions.dart' ('../../flutter/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart').
Try correcting the name to the name of an existing method, or defining a method named 'UnmodifiableListView'.
  List<SpawnedProcessRef> get processes => UnmodifiableListView(_processes);
                                           ^^^^^^^^^^^^^^^^^^^^
../../flutter/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart:407:34: Error: The method 'UnmodifiableListView' isn't defined for the class 'OpenFileList'.
 - 'OpenFileList' is from 'package:vm_service/src/dart_io_extensions.dart' ('../../flutter/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart').
Try correcting the name to the name of an existing method, or defining a method named 'UnmodifiableListView'.
  List<OpenFileRef> get files => UnmodifiableListView(_files);
                                 ^^^^^^^^^^^^^^^^^^^^
Failed to reactivate devtools 0.9.2: Failed to precompile devtools:devtools:
../../flutter/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart:276:33: Error: The method 'UnmodifiableListView' isn't defined for the class 'SpawnedProcess'.
 - 'SpawnedProcess' is from 'package:vm_service/src/dart_io_extensions.dart' ('../../flutter/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart').
Try correcting the name to the name of an existing method, or defining a method named 'UnmodifiableListView'.
  List<String> get arguments => UnmodifiableListView(_arguments);
                                ^^^^^^^^^^^^^^^^^^^^
../../flutter/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart:302:44: Error: The method 'UnmodifiableListView' isn't defined for the class 'SpawnedProcessList'.
 - 'SpawnedProcessList' is from 'package:vm_service/src/dart_io_extensions.dart' ('../../flutter/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart').
Try correcting the name to the name of an existing method, or defining a method named 'UnmodifiableListView'.
  List<SpawnedProcessRef> get processes => UnmodifiableListView(_processes);
                                           ^^^^^^^^^^^^^^^^^^^^
../../flutter/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart:407:34: Error: The method 'UnmodifiableListView' isn't defined for the class 'OpenFileList'.
 - 'OpenFileList' is from 'package:vm_service/src/dart_io_extensions.dart' ('../../flutter/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart').
Try correcting the name to the name of an existing method, or defining a method named 'UnmodifiableListView'.
  List<OpenFileRef> get files => UnmodifiableListView(_files);
                                 ^^^^^^^^^^^^^^^^^^^^
Failed to reactivate 1 package:
- devtools
```

</details>

<details>
<summary>Dart: Capture Logs</summary>

![image](https://user-images.githubusercontent.com/699655/93644034-7c90d800-fa01-11ea-8326-ee57874eb450.png)
![image](https://user-images.githubusercontent.com/699655/93644086-916d6b80-fa01-11ea-8b4f-af86af56bd11.png)

```
!! PLEASE REVIEW THIS LOG FOR SENSITIVE INFORMATION BEFORE SHARING !!

Dart Code extension: 3.14.1
Flutter extension: 3.14.1 (activated)

App: Visual Studio Code
Version: 1.49.1
Platform: mac

Workspace type: Flutter
Analyzer type: DAS
Multi-root?: false

Dart SDK:
    Loc: /Users/username/flutter/bin/cache/dart-sdk
    Ver: 2.9.2
Flutter SDK:
    Loc: /Users/username/flutter
    Ver: 1.20.4

HTTP_PROXY: undefined
NO_PROXY: undefined

Fri Sep 18 2020 [22:51:06 GMT+0200 (Central European Summer Time)] Log file started
[10:51:09 PM] [General] [Info] Spawning /Users/username/flutter/bin/cache/dart-sdk/bin/pub with args ["global","list"]
[10:51:09 PM] [CommandProcesses] [Info] (PROC 80475) Logging data for process...
[10:51:09 PM] [CommandProcesses] [Info] (PROC 80475) devtools 0.9.2
[10:51:09 PM] [CommandProcesses] [Info] (PROC 80475) exited (0, null)
[10:51:09 PM] [CommandProcesses] [Info] (PROC 80475) closed (0, null)
[10:51:09 PM] [DevTools] [Info] Spawning /Users/username/flutter/bin/cache/dart-sdk/bin/pub with args ["global","run","devtools","--machine","--try-ports","10","--allow-embedding"]
[10:51:09 PM] [DevTools] [Info] ..  with {"toolEnv":{"FLUTTER_HOST":"VSCode","PUB_ENVIRONMENT":"vscode.dart-code"}}
[10:51:09 PM] [DevTools] [Info]     PID: 75870
[10:51:09 PM] [DevTools] [Info] <== Precompiling executable...
[10:51:11 PM] [DevTools] [Error] Failed to precompile devtools:devtools:
[10:51:11 PM] [DevTools] [Error] Users/username/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart:276:33: Error: The method 'UnmodifiableListView' isn't defined for the class 'SpawnedProcess'.
 - 'SpawnedProcess' is from 'package:vm_service/src/dart_io_extensions.dart' ('Users/username/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart').
Try correcting the name to the name of an existing method, or defining a method named 'UnmodifiableListView'.
  List<String> get arguments => UnmodifiableListView(_arguments);
                                ^^^^^^^^^^^^^^^^^^^^
Users/username/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart:302:44: Error: The method 'UnmodifiableListView' isn't defined for the class 'SpawnedProcessList'.
 - 'SpawnedProcessList' is from 'package:vm_service/src/dart_io_extensions.dart' ('Users/username/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart').
Try correcting the name to the name of an existing method, or defining a method named 'UnmodifiableListView'.
  List<SpawnedProcessRef> get processes => UnmodifiableListView(_processes);
                                           ^^^^^^^^^^^^^^^^^^^^
Users/username/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart:407:34: Error: The method 'UnmodifiableListView' isn't defined for the class 'OpenFileList'.
 - 'OpenFileList' is from 'package:vm_service/src/dart_io_extensions.dart' ('Users/username/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart').
[10:51:11 PM] [DevTools] [Error] Try correcting the name to the name of an existing method, or defining a method named 'UnmodifiableListView'.
  List<OpenFileRef> get files => UnmodifiableListView(_files);
                                 ^^^^^^^^^^^^^^^^^^^^
[10:51:11 PM] [DevTools] [Error] 
[10:51:11 PM] [DevTools] [Info] Process terminated! 1, null
[10:51:11 PM] [General] [Error] Dart DevTools exited with code 1.
[10:51:48 PM] [General] [Info] Spawning /Users/username/flutter/bin/cache/dart-sdk/bin/pub with args ["global","activate","devtools"]
[10:51:48 PM] [CommandProcesses] [Info] (PROC 80753) Logging data for process...
[10:51:48 PM] [CommandProcesses] [Info] (PROC 80753) Package devtools is currently active at version 0.9.2.
[10:51:48 PM] [CommandProcesses] [Info] (PROC 80753) Resolving dependencies...
[10:51:50 PM] [CommandProcesses] [Info] (PROC 80753) + args 1.6.0
[10:51:50 PM] [CommandProcesses] [Info] (PROC 80753) + async 2.4.2 (2.5.0-nullsafety available)
+ browser_launcher 0.1.7
+ charcode 1.1.3 (1.2.0-nullsafety available)
+ collection 1.14.13 (1.15.0-nullsafety.2 available)
[10:51:50 PM] [CommandProcesses] [Info] (PROC 80753) 
+ convert 2.1.1
[10:51:50 PM] [CommandProcesses] [Info] (PROC 80753) 
+ crypto 2.1.5
+ devtools 0.9.2
[10:51:50 PM] [CommandProcesses] [Info] (PROC 80753) + devtools_server 0.9.2
+ devtools_shared 0.9.2
[10:51:50 PM] [CommandProcesses] [Info] (PROC 80753) 
+ http 0.12.2
+ http_multi_server 2.2.0
+ http_parser 3.1.4
[10:51:50 PM] [CommandProcesses] [Info] (PROC 80753) 
+ intl 0.16.1
+ logging 0.11.4
[10:51:50 PM] [CommandProcesses] [Info] (PROC 80753) 
+ meta 1.2.3 (1.3.0-nullsafety.2 available)
+ mime 0.9.7
[10:51:50 PM] [CommandProcesses] [Info] (PROC 80753) 
+ path 1.7.0 (1.8.0-nullsafety available)
[10:51:50 PM] [CommandProcesses] [Info] (PROC 80753) + pedantic 1.9.2 (1.10.0-nullsafety available)
+ shelf 0.7.9
+ shelf_proxy 0.1.0+7
[10:51:50 PM] [CommandProcesses] [Info] (PROC 80753) 
+ shelf_static 0.2.8
[10:51:50 PM] [CommandProcesses] [Info] (PROC 80753) + source_span 1.7.0 (1.8.0-nullsafety.1 available)
+ sse 3.5.0
[10:51:50 PM] [CommandProcesses] [Info] (PROC 80753) + stack_trace 1.9.5 (1.10.0-nullsafety available)
+ stream_channel 2.0.0 (2.1.0-nullsafety available)
[10:51:50 PM] [CommandProcesses] [Info] (PROC 80753) + string_scanner 1.0.5 (1.1.0-nullsafety available)
+ term_glyph 1.1.0 (1.2.0-nullsafety available)
[10:51:50 PM] [CommandProcesses] [Info] (PROC 80753) 
+ typed_data 1.2.0 (1.3.0-nullsafety.2 available)
+ usage 3.4.2
[10:51:50 PM] [CommandProcesses] [Info] (PROC 80753) 
+ uuid 2.2.2
[10:51:50 PM] [CommandProcesses] [Info] (PROC 80753) + vm_service 5.1.0
[10:51:50 PM] [CommandProcesses] [Info] (PROC 80753) + webkit_inspection_protocol 0.7.3
[10:51:50 PM] [CommandProcesses] [Info] (PROC 80753) Precompiling executables...
[10:51:51 PM] [CommandProcesses] [Info] (PROC 80753) Failed to precompile devtools:devtools:
[10:51:51 PM] [CommandProcesses] [Info] (PROC 80753) Users/username/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart:276:33: Error: The method 'UnmodifiableListView' isn't defined for the class 'SpawnedProcess'.
 - 'SpawnedProcess' is from 'package:vm_service/src/dart_io_extensions.dart' ('Users/username/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart').
Try correcting the name to the name of an existing method, or defining a method named 'UnmodifiableListView'.
  List<String> get arguments => UnmodifiableListView(_arguments);
                                ^^^^^^^^^^^^^^^^^^^^
Users/username/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart:302:44: Error: The method 'UnmodifiableListView' isn't defined for the class 'SpawnedProcessList'.
 - 'SpawnedProcessList' is from 'package:vm_service/src/dart_io_extensions.dart' ('Users/username/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart').
Try correcting the name to the name of an existing method, or defining a method named 'UnmodifiableListView'.
  List<SpawnedProcessRef> get processes => UnmodifiableListView(_processes);
[10:51:51 PM] [CommandProcesses] [Info] (PROC 80753) 
                                           ^^^^^^^^^^^^^^^^^^^^
Users/username/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart:407:34: Error: The method 'UnmodifiableListView' isn't defined for the class 'OpenFileList'.
[10:51:51 PM] [CommandProcesses] [Info] (PROC 80753) 
 - 'OpenFileList' is from 'package:vm_service/src/dart_io_extensions.dart' ('Users/username/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart').
[10:51:51 PM] [CommandProcesses] [Info] (PROC 80753) 
Try correcting the name to the name of an existing method, or defining a method named 'UnmodifiableListView'.
[10:51:51 PM] [CommandProcesses] [Info] (PROC 80753)   List<OpenFileRef> get files => UnmodifiableListView(_files);
                                 ^^^^^^^^^^^^^^^^^^^^
[10:51:51 PM] [CommandProcesses] [Info] (PROC 80753) exited (1, null)
[10:51:51 PM] [CommandProcesses] [Info] (PROC 80753) closed (1, null)
[10:51:51 PM] [General] [Error] Dart DevTools exited with code 1.

Package devtools is currently active at version 0.9.2.
Resolving dependencies...
+ args 1.6.0
+ async 2.4.2 (2.5.0-nullsafety available)
+ browser_launcher 0.1.7
+ charcode 1.1.3 (1.2.0-nullsafety available)
+ collection 1.14.13 (1.15.0-nullsafety.2 available)
+ convert 2.1.1
+ crypto 2.1.5
+ devtools 0.9.2
+ devtools_server 0.9.2
+ devtools_shared 0.9.2
+ http 0.12.2
+ http_multi_server 2.2.0
+ http_parser 3.1.4
+ intl 0.16.1
+ logging 0.11.4
+ meta 1.2.3 (1.3.0-nullsafety.2 available)
+ mime 0.9.7
+ path 1.7.0 (1.8.0-nullsafety available)
+ pedantic 1.9.2 (1.10.0-nullsafety available)
+ shelf 0.7.9
+ shelf_proxy 0.1.0+7
+ shelf_static 0.2.8
+ source_span 1.7.0 (1.8.0-nullsafety.1 available)
+ sse 3.5.0
+ stack_trace 1.9.5 (1.10.0-nullsafety available)
+ stream_channel 2.0.0 (2.1.0-nullsafety available)
+ string_scanner 1.0.5 (1.1.0-nullsafety available)
+ term_glyph 1.1.0 (1.2.0-nullsafety available)
+ typed_data 1.2.0 (1.3.0-nullsafety.2 available)
+ usage 3.4.2
+ uuid 2.2.2
+ vm_service 5.1.0
+ webkit_inspection_protocol 0.7.3
Precompiling executables...


Failed to precompile devtools:devtools:
Users/username/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart:276:33: Error: The method 'UnmodifiableListView' isn't defined for the class 'SpawnedProcess'.
 - 'SpawnedProcess' is from 'package:vm_service/src/dart_io_extensions.dart' ('Users/username/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart').
Try correcting the name to the name of an existing method, or defining a method named 'UnmodifiableListView'.
  List<String> get arguments => UnmodifiableListView(_arguments);
                                ^^^^^^^^^^^^^^^^^^^^
Users/username/.pub-cache/hosted/pub.dartlang.org/vm_service-5.1.0/lib/src/dart_io_extensions.dart:302:44: Error: The method 'UnmodifiableListView' isn't defined for the class 'SpawnedProcessList'.
 - 'SpawnedProcessList' is from 'package:vm_se…
Fri Sep 18 2020 [22:52:13 GMT+0200 (Central European Summer Time)] Log file ended
```

</details>

**Comment by kenzieschmoll** [comment]
Do you still see this issue if you upgrade your Flutter?
cc @bkonyi

**Comment by Jomik** [comment]
> Do you still see this issue if you upgrade your Flutter?
> cc @bkonyi

I just ran `flutter upgrade` prior to this issue. Is v1.20.4 not the latest (stable)?

**Comment by devoncarew** [comment]
@bkonyi - this may be due to the recent publish of package:vm_service.

**Comment by bkonyi** [comment]
Yes, this should be resolved with 5.1.0+1.

**Comment by Jomik** [comment]
> Yes, this should be resolved with 5.1.0+1.
Which it should automatically resolve to, when getting dependencies, so I just run a repair or something?

**Comment by Jomik** [comment]
Yeah, this works fine now 👍
